airflow/executors/kubernetes_executor.py

```diff
@@ -187,25 +190,35 @@ def process_status(
         self,
         pod_id: str,
         namespace: str,
-        status: str,
+        status: V1PodStatus,
         annotations: Dict[str, str],
         resource_version: str,
         event: Any,
     ) -> None:
         """Process status response"""
-        if status == 'Pending':
-            if event['type'] == 'DELETED':
+        pod_status = status.phase
+        if pod_status == 'Pending':
+            # Check container statuses
+            container_statuses = status.container_statuses
+            init_container_statuses = status.init_container_statuses
+            if container_statuses and self._container_image_pull_err(container_statuses):
+                self.log.info('Event: Failed to start pod %s, a container has an ErrImagePull', pod_id)
+                self.watcher_queue.put((pod_id, namespace, State.FAILED, annotations, resource_version))
```

Comment:
While this marks the tasks as failed, the pod remains in 'Pending' state and thus not deleted. Looking for a way to change the Pod status

Reply:
@ephraimbuddy I don't think you can change the pod status from inside of Airflow.

Reply:
Yeah, which is why #15263 deletes the pod. I think that is all we can do, unfortunately.

That said, should we be failing these after the first pull failure? It is certainly something that could self-resolve and we haven't actually started the task yet 🤷‍♂️. I'm thinking a timeout is a more general catch-all for pending issues.

Reply:
Interesting!!